datafusion/core/tests/fuzz_cases/window_fuzz.rs

```diff
@@ -146,6 +163,7 @@ async fn window_bounded_window_random_comparison() -> Result<()> {
 // This tests whether we can generate bounded window results for each input
 // batch immediately for causal window frames.
 #[tokio::test(flavor = "multi_thread", worker_threads = 16)]
+#[ignore = "Non-nullable window count(*) can currently return NULL value in some edge cases. See tmp_test()"]
```

Comment:
Only in our branch? What's special about it?

Reply:
So on our fork we have a commit:

- "Physical-plan counterpart for DF https://github.com/apache/datafusion/pull/11256" (https://github.com/sdf-labs/arrow-datafusion/commit/d595069bf95db95ad14ce1826d482aee13a526e6)

It breaks tests and uncovers bugs. Fixing most of the test is easy (just update expected value), but there are 2+ problematic ones:  `bounded_window_causal_non_causal`, `window_bounded_window_random_comparison`. They fail (`Column 'count' is declared as non-nullable but contains null values`) because the schema returned marks count as null, but implementation (incorrectly) returns null for queries like `SELECT count(x) OVER (ORDER BY x NULLS FIRST ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING) AS c FROM (VALUES (1), (2), (NULL), (3)) t(x)`

- In DF upstream this was fixed in https://github.com/apache/datafusion/pull/11989  but it's not easily cherry-pickable (i tried)
- also,  https://github.com/apache/datafusion/issues/13065 suggests it's potentially disruptive change

Reply:
this is a TMP commit we will drop during next update. there is a test making sure we don't forget, even if we don't read commit titles

Reply:
@findepi, I bow in awe for your catching, understanding and illuminating the consequences of that seemingy-innocent count() nullability tweak! (in DF 39, if I recall?) 
My fix in https://github.com/sdf-labs/arrow-datafusion/commit/d595069bf95db95ad14ce1826d482aee13a526e6 was sheer luck -- both in finding it and not in triggering more problems, which you uncovered were lurking there!

Reply:
@vgapeyev i just followed failing test cases, nothing special.
Then i realized the upstream has equivalent of your fix, so i `git bisect`-ed to find the commit which did it (grepping `git log` for "count" or "window" couldn't find the right commit).